File: tests/exercises/echo-function/evaluation/evaluator.py
# noinspection PyUnresolvedReferences
from evaluation_utils import EvaluationResult, Message


def evaluate(actual):
    correct = actual == "correct"
    return EvaluationResult(correct, "correct", actual, [Message("Hallo")])


def evaluate_value(context):
    return EvaluationResult(context.expected == context.actual, context.expected, context.actual, [Message("Hallo")])


def evaluate_value_dsl(context):
    return EvaluationResult(
        result=context.expected == context.actual,
        messages=[Message("Hallo")],
        dsl_expected="{5, 5}",
        dsl_actual="{4, 4}"
    )


def evaluate_runtime_crash(context):
    return len(context) / 0


def evaluate_sum(actual, the_sum):
    correct = the_sum == 10
    return EvaluationResult(correct, "correct", actual, [Message("Hallo")])

File: tests/exercises/echo-function/evaluation/Evaluator.kt
class Evaluator {
    companion object {
        @JvmStatic
        fun evaluate(actual: Any?): EvaluationResult {
            return EvaluationResult.Builder(result = "correct" == actual,
                    readableExpected = actual.toString(),
                    readableActual = actual?.toString() ?: "")
                    .withMessage(EvaluationResult.Message("Hallo"))
                    .build()
        }

        @JvmStatic
        fun evaluateValue(context: Map<String, Any>): EvaluationResult {
            return EvaluationResult.Builder(result = context["expected"] == context["actual"],
                    readableExpected = context["expected"].toString(),
                    readableActual = context["actual"]?.toString() ?: "")
                    .withMessage(EvaluationResult.Message("Hallo"))
                    .build()
        }

        @JvmStatic
        fun evaluateValueDsl(context: Map<String, Any>): EvaluationResult {
            return EvaluationResult.Builder(result = context["expected"] == context["actual"],
                    dslExpected = "{5, 5}",
                    dslActual = "{4, 4}")
                    .withMessage(EvaluationResult.Message("Hallo"))
                    .build()
        }
    }
}
